modules/multi-architecture-modify-machine-set-gcp.adoc

```diff
@@ -0,0 +1,126 @@
+
+.Procedure
+* Create and modify a compute machine set, this will control the ARM64 compute nodes in your cluster. 
++
+[source,terminal]
+----
+$ oc create -f gcp-arm64-machine-set-0.yaml
+----
++
+--
+.Sample GCP YAML compute machine set to deploy an ARM64 compute node
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_id> <1>
+  name: <infrastructure_id>-w-a
+  namespace: openshift-machine-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+      machine.openshift.io/cluster-api-machineset: <infrastructure_id>-w-a
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        machine.openshift.io/cluster-api-cluster: <infrastructure_id>
+        machine.openshift.io/cluster-api-machine-role: <role> <2>
+        machine.openshift.io/cluster-api-machine-type: <role>
+        machine.openshift.io/cluster-api-machineset: <infrastructure_id>-w-a
+    spec:
+      metadata:
+        labels:
+          node-role.kubernetes.io/<role>: ""
+      providerSpec:
+        value:
+          apiVersion: gcpprovider.openshift.io/v1beta1
+          canIPForward: false
+          credentialsSecret:
+            name: gcp-cloud-credentials
+          deletionProtection: false
+          disks:
+          - autoDelete: true
+            boot: true
+            image: <path_to_image> <3>
+            labels: null
+            sizeGb: 128
+            type: pd-ssd
+          gcpMetadata: <4>
+          - key: <custom_metadata_key>
+            value: <custom_metadata_value>
+          kind: GCPMachineProviderSpec
+          machineType: n1-standard-4 <5>
+          metadata:
+            creationTimestamp: null
+          networkInterfaces:
+          - network: <infrastructure_id>-network
+            subnetwork: <infrastructure_id>-worker-subnet
+          projectID: <project_name> <6>
+          region: us-central1 <7>
+          serviceAccounts:
+          - email: <infrastructure_id>-w@<project_name>.iam.gserviceaccount.com
+            scopes:
+            - https://www.googleapis.com/auth/cloud-platform
+          tags:
+            - <infrastructure_id>-worker
+          userDataSecret:
+            name: worker-user-data
+          zone: us-central1-a
+----
+<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the OpenShift CLI installed, you can obtain the infrastructure ID by running the following command:
++
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+<2> Specify the role node label to add.
+<3> Specify the path to the image that is used in current compute machine sets. 
+// the OCP docs reference the GCP Marketplace image, not sure if that is supported on ARM64
```

Comment:
i don't believe the market place images are supported, but I guess the same query that we use to get AMis for AWS can be used here to fetch the appropriate gcp images from the rhcos.json file in the installer

Reply:
Is there a specific command that can do that similar to AWS for GCP? Or is it something completely different?

Reply:
it should be similar to what we have for aws today: https://docs.openshift.com/container-platform/4.13/post_installation_configuration/multi-architecture-configuration.html#multi-architecture-modify-machine-set-aws_multi-architecture-configuration.

The command to get the arm64 image for gcp will have to be slightly tweaked, so it should be:
```
oc get configmap/coreos-bootimages /
	  -n openshift-machine-config-operator /
	  -o jsonpath='{.data.stream}' | jq /
	  -r '.architectures.<arch>.images.gcp.name'
```
Although I'm not entirely sure the deployment directly uses this image as a bootimage or creates a temporary image. @aleskandro can you or your QE colleagues verify this?

Reply:
Yes, the command above is fine.

The path to the image takes this form:
```php
projects/${project}/global/images/${image_name}
```

with image_name as above and project as in  

```
oc get configmap/coreos-bootimages / -n openshift-machine-config-operator / -o jsonpath='{.data.stream}' | jq / -r '.architectures.<arch>.images.gcp.project'
```

In this case arch is the GNU arch (aarch64, x86_64, others...)

Reply:
nit in the command: `/` vs `\`